WordPress/src/test/java/org/wordpress/android/ui/sitecreation/verticals/NewSiteCreationVerticalsViewModelTest.kt

```diff
@@ -0,0 +1,395 @@
+package org.wordpress.android.ui.sitecreation.verticals
+
+import android.arch.core.executor.testing.InstantTaskExecutorRule
+import android.arch.lifecycle.LiveData
+import android.arch.lifecycle.Observer
+import com.nhaarman.mockito_kotlin.any
+import com.nhaarman.mockito_kotlin.anyOrNull
+import com.nhaarman.mockito_kotlin.firstValue
+import com.nhaarman.mockito_kotlin.lastValue
+import com.nhaarman.mockito_kotlin.secondValue
+import com.nhaarman.mockito_kotlin.thirdValue
+import com.nhaarman.mockito_kotlin.times
+import com.nhaarman.mockito_kotlin.verify
+import com.nhaarman.mockito_kotlin.whenever
+import kotlinx.coroutines.experimental.CoroutineScope
+import kotlinx.coroutines.experimental.Dispatchers
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.ArgumentCaptor
+import org.mockito.Mock
+import org.mockito.junit.MockitoJUnitRunner
+import org.wordpress.android.fluxc.Dispatcher
+import org.wordpress.android.fluxc.model.vertical.SegmentPromptModel
+import org.wordpress.android.fluxc.model.vertical.VerticalModel
+import org.wordpress.android.fluxc.store.VerticalStore.FetchSegmentPromptError
+import org.wordpress.android.fluxc.store.VerticalStore.FetchVerticalsError
+import org.wordpress.android.fluxc.store.VerticalStore.OnSegmentPromptFetched
+import org.wordpress.android.fluxc.store.VerticalStore.OnVerticalsFetched
+import org.wordpress.android.fluxc.store.VerticalStore.VerticalErrorType.GENERIC_ERROR
+import org.wordpress.android.test
+import org.wordpress.android.ui.sitecreation.usecases.FetchSegmentPromptUseCase
+import org.wordpress.android.ui.sitecreation.usecases.FetchVerticalsUseCase
+import org.wordpress.android.ui.sitecreation.verticals.NewSiteCreationVerticalsViewModel.VerticalsContentState.CONTENT
+import org.wordpress.android.ui.sitecreation.verticals.NewSiteCreationVerticalsViewModel.VerticalsListItemUiState.VerticalsFetchSuggestionsErrorUiState
+import org.wordpress.android.ui.sitecreation.verticals.NewSiteCreationVerticalsViewModel.VerticalsListItemUiState.VerticalsModelUiState
+import org.wordpress.android.ui.sitecreation.verticals.NewSiteCreationVerticalsViewModel.VerticalsUiState
+
+const val SEGMENT_ID = 1L
+const val ZERO_DELAY = 0
+const val EMPTY_STRING = ""
+
+const val DUMMY_HINT = "dummyHint"
+const val DUMMY_TITLE = "dummyTitle"
+const val DUMMY_SUBTITLE = "dummySubtitle"
+
+const val FIRST_MODEL_QUERY = "success_1"
+const val SECOND_MODEL_QUERY = "success_2"
+const val ERROR_MODEL_QUERY = "error"
+
+const val FIRST_MODEL_NAME = "firstModel"
+const val FIRST_MODEL_ID = "1"
+const val SECOND_MODEL_NAME = "secondModel"
+const val SECOND_MODEL_ID = "2"
+
+val SUCCESSFUL_HEADER_PROMPT_FETCHED = OnSegmentPromptFetched(
```

Comment:
Fwiw I wasn't talking about these `val`s in my initial review, I was talking about the ones like `DUMMY_TITLE` when I said we might want to move them out of the class because I think these ones make more sense in the class. Sorry if my comment wasn't clear. _This is assuming you moved them due to my comment and not because you wanted to in which case you can ignore this comment._

Reply:
I moved them so I could use capital letters with underscores to highlight they are "constant" fields. However, if you prefer to move them back, just let me know.

Reply:
Nope, that's totally fine, I just wanted to make sure my previous comment was clear. 👍